cwbi-auth-http-client/src/main/java/hec/army/usace/hec/cwbi/auth/http/client/RefreshTokenRequestBuilder.java

```diff
@@ -0,0 +1,52 @@
+package hec.army.usace.hec.cwbi.auth.http.client;
+
+import hec.army.usace.hec.cwbi.auth.http.client.token.fluentbuilders.RefreshTokenRequestFluentBuilder;
+import hec.army.usace.hec.cwbi.auth.http.client.token.fluentbuilders.TokenRequestFluentBuilder;
+import java.io.IOException;
+import java.util.Objects;
+import mil.army.usace.hec.cwms.http.client.ApiConnectionInfoBuilder;
+import mil.army.usace.hec.cwms.http.client.HttpRequestBuilderImpl;
+import mil.army.usace.hec.cwms.http.client.HttpRequestResponse;
+import mil.army.usace.hec.cwms.http.client.auth.OAuth2Token;
+import mil.army.usace.hec.cwms.http.client.request.HttpRequestExecutor;
+
+public final class RefreshTokenRequestBuilder implements RefreshTokenRequestFluentBuilder {
```

Comment:
Where is the business that checks if the token expiration is close and it needs a refresh?

Reply:
It was no longer needed. OkHttpClient has an authenticator process that handles expired tokens. If expired, it will auto-refresh the token behind the scenes. That was added to cwms-http-client per Adam's request for re-use. This also allows the user to only have to enter their CAC pin once per session, and it will auto-refresh from then on.